bitcoin/src/blockdata/transaction.rs

```diff
@@ -713,28 +713,32 @@ impl Transaction {
     }
 
     /// Returns the size of this transaction excluding the witness data.
-    pub fn strippedsize(&self) -> usize {
-        let mut input_size = 0;
+    #[deprecated(since = "0.31.0", note = "Use Transaction::stripped_size() instead")]
+    pub fn strippedsize(&self) -> usize { Self::stripped_size(self).to_wu() as usize }
+
+    /// Returns the size of this transaction excluding the witness data.
+    pub fn stripped_size(&self) -> Weight {
+        let mut input_size: Weight = Weight::ZERO;
         for input in &self.input {
-            input_size += TxIn::BASE_WEIGHT.to_wu() as usize
-                + VarInt(input.script_sig.len() as u64).len()
-                + input.script_sig.len();
+            input_size += TxIn::BASE_WEIGHT
+                + Weight::from_wu_usize(VarInt(input.script_sig.len() as u64).len())
+                + Weight::from_wu_usize(input.script_sig.len());
         }
-        let mut output_size = 0;
+        let mut output_size = Weight::ZERO;
         for output in &self.output {
-            output_size += 8 + // value
-                VarInt(output.script_pubkey.len() as u64).len() +
-                output.script_pubkey.len();
+            output_size += Weight::from_wu(8)+ // value
+                Weight::from_wu_usize(VarInt(output.script_pubkey.len() as u64).len()) +
+                Weight::from_wu_usize(output.script_pubkey.len());
         }
-        let non_input_size =
+        let non_input_size: Weight =
         // version:
-        4 +
+        Weight::from_wu(4)+
```

Comment:
And this magic number. (This one is probably a 4 bytes for a `u32`, thats pretty clear but a const would hurt.)

Reply:
Good idea, but I feel like there is enough commits going on in this PR.  I'll look into this for a future PR.